syntax = "proto3";

package contracts.consumerstore;
option go_package = "github.com/makkalot/eskit/generated/grpc/go/consumerstore";

import "common/originator.proto";
import "google/api/annotations.proto";


message ListConsumersRequest {
}

message ListConsumersResponse {
    repeated GetAppLogConsumeResponse consumers = 1;
}

message AppLogConsumeRequest {
    // the consumer id
    string consumer_id = 1;
    string offset = 2;
}

message AppLogConsumeResponse {
}

message GetAppLogConsumeRequest {
    string consumer_id = 1;
}


message GetAppLogConsumeResponse {
    string consumer_id = 1;
    string offset = 2;
}

message HealthRequest {
}

message HealthResponse {
    string message = 1;
}

service ConsumerService {

    rpc Healtz (HealthRequest) returns (HealthResponse) {
        option (google.api.http) = {
            get: "/v1/healtz"
        };
    }

    rpc LogConsume (AppLogConsumeRequest) returns (AppLogConsumeResponse) {
    }

    rpc GetLogConsume (GetAppLogConsumeRequest) returns (GetAppLogConsumeResponse) {
    }

    rpc List (ListConsumersRequest) returns (ListConsumersResponse) {

    }
}